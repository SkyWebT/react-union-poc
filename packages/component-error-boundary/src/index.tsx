import React from 'react';

class ErrorBoundary extends React.Component<{}, { hasError: boolean }> {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    // Update state so the next render will show the fallback UI.
    return { hasError: true };
  }

  componentDidCatch(error, errorInfo) {
    // You can also log the error to an error reporting service
    console.log(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return (
        <div style={{ backgroundColor: 'red' }}>
          <h2>Something went wrong</h2>
          <p>Our engineer team is aware of this issue. </p>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
