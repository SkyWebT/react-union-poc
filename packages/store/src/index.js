export { create ,actions} from './store';
