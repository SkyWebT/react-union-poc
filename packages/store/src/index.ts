export { create, actions, AppState } from './store';
