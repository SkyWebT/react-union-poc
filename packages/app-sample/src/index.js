import React from 'react';
import { Provider } from 'react-redux';
import { justRender } from 'react-union';
import { create } from '@skytv/store';

import Root from './components/Root';

const store = create();

const app = (
	<Provider store={store}>
		<Root />
	</Provider>
);

if ('Liferay' in window) {
	window.Liferay.on('allPortletsReady', function() {
		console.log('allPortletsReady !!!!!!!');
		justRender(app);
	});
} else {
	justRender(app);
}
