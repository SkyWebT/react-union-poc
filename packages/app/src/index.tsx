import React from 'react';
import { Provider } from 'react-redux';
import { justRender } from 'react-union';

import { create } from '@sky/store';

import Root from './Root';

const store = create();

const app = (
  <Provider store={store}>
    <Root />
  </Provider>
);

declare global {
  interface Window {
    Liferay: any;
  }
}
if ('Liferay' in window) {
  window.Liferay.on('allPortletsReady', function() {
    console.log('allPortletsReady !!!!!!!');
    justRender(app);
  });
} else {
  justRender(app);
}
