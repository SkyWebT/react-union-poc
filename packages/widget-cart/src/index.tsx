import React from 'react';
import { connect } from 'react-redux';
import { createSelector, PayloadActionCreator } from 'redux-starter-kit';

import { actions, AppState, T_Product } from '@skytv/store';

const selectSelected = (state: AppState) => Object.values(state.products).filter(p => p.selected);

const selector = createSelector(
  selectSelected,
  products => ({
    total: products.reduce((acc, cur) => {
      acc += cur.price;
      return acc;
    }, 0),
    products,
  })
);

interface Props {
  products: T_Product[];
  total: number;
  toggle: PayloadActionCreator<string>;
}
const Root: React.FC<Props> = ({ products, total, toggle }) => {
  const removeFromCart = (id: string) => {
    toggle(id);
  };
  return (
    <div style={{ border: '1px solid grey', padding: 20, width: '40vw' }}>
      <h2>Cart </h2>
      {products.length > 0 ? (
        products.map(({ id, name, price }) => (
          <div key={id} style={{ display: 'flex', justifyContent: 'space-between' }}>
            <h3>
              {name} - ${price}
            </h3>
            <div>
              <button onClick={() => removeFromCart(id)}>remove</button>
            </div>
          </div>
        ))
      ) : (
        <div> empty </div>
      )}
      <h3>Total: ${total}</h3>
    </div>
  );
};

export default connect(
  selector,
  actions.prodducts
)(Root);
