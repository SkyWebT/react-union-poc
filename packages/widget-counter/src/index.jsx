import { useSelector, useDispatch } from 'react-redux';
import React from 'react';
import { actions } from '@skytv/store';
import style from './style.css';

const Root = () => {
  const { counter } = useSelector(s => s.counter);
  const d = useDispatch();
  const inc = () => {
    d(actions.counter.increment());
  };
  const dec = () => {
    d(actions.counter.decrement());
  };
  return (
    <div className={style.counter}>
      <div>hello typescript counter</div>
      <div>{counter}</div>
      <button onClick={inc}>+++</button>
      <button onClick={dec}>---</button>
    </div>
  );
};

export default Root;
