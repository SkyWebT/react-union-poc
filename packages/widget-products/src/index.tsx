import React from 'react';
import { connect } from 'react-redux';
import { PayloadActionCreator } from 'redux-starter-kit';

import { actions, AppState, T_Product } from '@sky/store';

interface Props {
  products: T_Product[];
  toggle: PayloadActionCreator<string>;
}
export const Products: React.FC<Props> = ({ products, toggle }) => {
  const addToCart = (id: string) => {
    toggle(id);
  };
  return (
    <div style={{ border: '1px solid grey', padding: 20, width: '40vw' }}>
      <h2>Product list</h2>

      {products.map(({ id, name, price, selected }) => (
        <div key={id} style={{ display: 'flex', justifyContent: 'space-between' }}>
          <h3>
            {name} - ${price}
          </h3>
          <div>
            <button disabled={selected} onClick={() => addToCart(id)}>
              buy
            </button>
          </div>
        </div>
      ))}
    </div>
  );
};

export default connect(
  (state: AppState) => ({ products: Object.values(state.products) }),
  actions.prodducts
)(Products);
