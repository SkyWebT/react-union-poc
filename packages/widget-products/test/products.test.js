import React from 'react';
import { Products } from '../src';
import { render, fireEvent } from '@testing-library/react';

describe('<Root />', () => {
  const state = {
    '1': {
      id: '1',
      name: 'SKY Starter',
      price: 125,
      selected: false,
    },
    '2': {
      id: '2',
      name: 'SKY Entertainment',
      price: 35,
      selected: false,
    },
    '3': {
      id: '3',
      name: 'SKY Sport',
      price: 99,
      selected: false,
    },
  };
  const requiredProps = {
    products: Object.values(state),
    toggle: id => {
      state[id].selected = !state[id].selected;
    },
  };

  it('should match snapshot', () => {
    const { asFragment } = render(<Products {...requiredProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should have buttons', () => {
    const { container } = render(<Products {...requiredProps} />);
    expect(container.getElementsByTagName('button').length).toEqual(Object.values(state).length);
  });
  it('should disabled button once clicked', () => {
    const { container, rerender } = render(<Products {...requiredProps} />);
    const button = container.getElementsByTagName('button')[0];

    expect(button.disabled).toEqual(false);
    fireEvent.click(button);
    rerender(<Products {...requiredProps} />);
    expect(button.disabled).toEqual(true);
  });
});
