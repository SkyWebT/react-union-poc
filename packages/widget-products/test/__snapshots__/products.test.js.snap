// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="border: 1px solid grey; padding: 20px; width: 40vw;"
  >
    <h2>
      Product list
    </h2>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <h3>
        SKY Starter - $125
      </h3>
      <div>
        <button>
          buy
        </button>
      </div>
    </div>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <h3>
        SKY Entertainment - $35
      </h3>
      <div>
        <button>
          buy
        </button>
      </div>
    </div>
    <div
      style="display: flex; justify-content: space-between;"
    >
      <h3>
        SKY Sport - $99
      </h3>
      <div>
        <button>
          buy
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
